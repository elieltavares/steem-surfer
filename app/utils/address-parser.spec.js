/* eslint-disable */

import addressParser from './address-parser';

describe('Address parser', () => {
  it('filter', () => {
    const input = `/trending`;
    expect(addressParser(input)).toMatchSnapshot();
  });

  it('filter + tag', () => {
    const input = `/trending/esteem`;
    expect(addressParser(input)).toMatchSnapshot();
  });

  it('author', () => {
    const input = `/@good-karma`;
    expect(addressParser(input)).toMatchSnapshot();
  });

  it('author + section', () => {
    const input = `/@good-karma/blog`;
    expect(addressParser(input)).toMatchSnapshot();
  });

  it('author + section', () => {
    const input = `/@good-karma/comments`;
    expect(addressParser(input)).toMatchSnapshot();
  });

  it('author + section', () => {
    const input = `/@good-karma/replies`;
    expect(addressParser(input)).toMatchSnapshot();
  });

  it('author + section', () => {
    const input = `/@good-karma/wallet`;
    expect(addressParser(input)).toMatchSnapshot();
  });

  it('author + section', () => {
    const input = `/@good-karma/feed`;
    expect(addressParser(input)).toMatchSnapshot();
  });

  it('post', () => {
    const input = `https://steemit.com/esteem/@good-karma/esteem-search-is-now-opensource-hivemind-plugin-58f8a8bad0598est`;
    expect(addressParser(input)).toMatchSnapshot();
  });

  it('post', () => {
    const input = `/@good-karma/esteem-search-is-now-opensource-hivemind-plugin-58f8a8bad0598est`;
    expect(addressParser(input)).toMatchSnapshot();
  });
});
