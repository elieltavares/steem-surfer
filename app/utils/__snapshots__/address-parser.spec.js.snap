// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address parser author + section 1`] = `
Object {
  "author": "good-karma",
  "filter": "blog",
  "path": "/@good-karma/blog",
  "type": "author",
}
`;

exports[`Address parser author + section 2`] = `
Object {
  "author": "good-karma",
  "filter": "comments",
  "path": "/@good-karma/comments",
  "type": "author",
}
`;

exports[`Address parser author + section 3`] = `
Object {
  "author": "good-karma",
  "filter": "replies",
  "path": "/@good-karma/replies",
  "type": "author",
}
`;

exports[`Address parser author + section 4`] = `
Object {
  "author": "good-karma",
  "filter": "wallet",
  "path": "/@good-karma/wallet",
  "type": "author",
}
`;

exports[`Address parser author + section 5`] = `
Object {
  "author": "good-karma",
  "filter": "feed",
  "path": "/@good-karma/feed",
  "type": "author",
}
`;

exports[`Address parser author 1`] = `
Object {
  "author": "good-karma",
  "filter": undefined,
  "path": "/@good-karma",
  "type": "author",
}
`;

exports[`Address parser filter + tag 1`] = `
Object {
  "filter": "trending",
  "path": "/trending/esteem",
  "tag": "esteem",
  "type": "filter",
}
`;

exports[`Address parser filter 1`] = `
Object {
  "filter": "trending",
  "path": "/trending",
  "tag": undefined,
  "type": "filter",
}
`;

exports[`Address parser post 1`] = `
Object {
  "author": "good-karma",
  "permlink": "esteem-search-is-now-opensource-hivemind-plugin-58f8a8bad0598est",
  "type": "post",
}
`;

exports[`Address parser post 2`] = `
Object {
  "author": "good-karma",
  "permlink": "esteem-search-is-now-opensource-hivemind-plugin-58f8a8bad0598est",
  "type": "post",
}
`;
